### PR TITLE
feat: Add prototxt script

### PR DESCRIPTION
**Pull request by mkmik**
This PR adds a prototxt script and accompanying documentation.

Rationale: help developers deal with binary protobuf files.

### CODE REVIEW

File: docs/protobuf.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+$ cat /tmp/iox-data/1/foobar_weather/rules.pb | hexdump -C
+00000000  0a 0e 66 6f 6f 62 61 72  5f 77 65 61 74 68 65 72  |..foobar_weather|
+00000010  12 15 0a 13 1a 11 25 59  2d 25 6d 2d 25 64 20 25  |......%Y-%m-%d %|
+00000020  48 3a 30 30 3a 30 30 3a  0a 08 c0 84 3d 1a 04 08  |H:00:00:....=...|
+00000030  02 1a 00                                          |...|
+```
+
+## Textpb
+
+The textual protobuf encoding (aka `textpb`) may be unfamiliar to most people. (TODO find a link to some textpb spec or doc)
```
Comment (tustvold) [inline docs/protobuf.md:62]: I'm not sure there is one, I don't think it is standardised

**Comment by tustvold** [comment]
I don't object to textpb, but I wonder if the JSON output, which is actually standardised, might be a better recommendation?

**Comment by mkmik** [comment]
> I don't object to textpb, but I wonder if the JSON output, which is actually standardised, might be a better recommendation?

It is/was used widely internally at google, but I think there is no strong push to standardize it, since there is jsonpb; unfortunately the tooling is crap.

I'm not a fan, but that's the simplest way we can get a textual format without finding/writing/installing extra tools.

I'm still baffled by the lack of good jsonpb tooling in tools like prototool or buf etc.

I will replace this crap once I stumble upon (or cobble together) a better tool.

File: docs/protobuf.md (hunk 1 above)
Comment (mkmik) [reply docs/protobuf.md]: removed the TODO; as it's unlikely it will ever be published

**Review by tustvold — APPROVED** [review]

File: docs/protobuf.md [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+```console
+$ ./scripts/prototxt types | grep DatabaseRules
+influxdata.iox.management.v1.DatabaseRules
+```
+
+You can now decode the binary protobuf into a textual form you can inspect and/or edit:
+
+```console
+$ ./scripts/prototxt decode influxdata.iox.management.v1.DatabaseRules \
```
Comment (alamb) [inline docs/protobuf.md:30]: this is awesome -- thank you @mkmik !

File: docs/protobuf.md [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  }
+  parts: <
+    time: "%Y-%m-%d %H:00:00"
+  >
+  parts <
+    time: "%Y-%m-%d %H:00:00"
+  >
+```
+
+(echoes of a distant past, when JSON wasn't a thing yet).
```
Comment (alamb) [inline docs/protobuf.md:119]: 😆

**Pull request merged**